### PR TITLE
fix: disableOrphans should not enable procedure callers if their associated definitions are disabled

### LINKED ISSUE #7154: disable-top-blocks should not enable procedure callers if their associated definitions are disabled
**Describe the bug**

Procedure callers are supposed to be disabled if their associated definitions are disabled. However, this plugin automatically enables procedure caller blocks when they are attached to a start block, even if they are supposed to be disabled.

**To Reproduce**

1. Open the playground file, and add the follow code at the bottom of the `start` function.
```js
workspace.addChangeListener(Blockly.Events.disableOrphans);
```
3. Drag two procedure definitions into the workspace.
4. Disable the second one.
5. Drag a caller for the second one into the workspace.
6. Observe how it is disabled.
7. Attach the caller to the first procedure.
8. Observe how it is incorrectly enabled (incorrectly because its definition is disabled).

image

**Expected behavior**

Procedure callers should remain disabled if their definitions are disabled.

image

## To Fix

1) Run through the reproduction steps and make sure you can reproduce the issue.
2) In the [disableOrphans](https://github.com/google/blockly/blob/344861698e65c3fc5f05db000c01cff2155f56ff/core/events/utils.ts#L542) listener check if the block [isProcedureBlock](https://github.com/google/blockly/blob/344861698e65c3fc5f05db000c01cff2155f56ff/core/interfaces/i_procedure_block.ts#L19). If it is, make sure that its procedure model is enabled before enabling it.
1) Run the reproduction steps again and make sure the issue doesn't reproduce.

### PR DESCRIPTION
**Pull request by Taru-Shukla**
## The basics

- [x] I [validated my changes](https://developers.google.com/blockly/guides/contribute/core#making_and_verifying_a_change)

### Resolves

disable-top-blocks should not enable procedure callers if their associated definitions are disabled
fixes #7154

### Proposed Changes

Using the Type Guard in Loop: Inside the loop that iterates over each child block, the code uses the isProcedureBlock function to determine if child is a procedure block.

```
if(isProcedureBlock(child) && !child.getProcedureModel().getEnabled()){
  continue;
}
```
`isProcedureBlock(child)`: Checks if the current child block is a procedure block.
`!child.getProcedureModel().getEnabled()`: Checks if the procedure model corresponding to this block is not enabled.
If both conditions are true, it skips enabling the current child block by using continue; to move to the next iteration of the loop.

### Reason for Changes

This condition does two things:

First, it checks if the child block is a procedure block using the isProcedureBlock function.
Second, if it's a procedure block, it checks if the procedure model for that block is enabled using `child.getProcedureModel().getEnabled()`. If the model isn't enabled, the loop continues to the next child block without enabling this particular child.
This ensures that if a block is a procedure block, it will only be enabled if its corresponding procedure model is enabled.

### CODE REVIEW

**Comment by rachel-fenichel** [comment]
Hi @Taru-Shukla! Did you mean to make this a draft pull request? If it's ready for review, please click the "Ready for review" button so it gets assigned to a member of our team.

File: tests/playgrounds/advanced_playground.html [hunk 1]
```diff
@@ -22,6 +22,7 @@
       function start() {
         setBackgroundColour();
         initPlayground();
+        workspace.addChangeListener(Blockly.Events.disableOrphans);
```
Comment (maribethb) [inline tests/playgrounds/advanced_playground.html:25]: We only needed this change in order to test the code you wrote above. We don't want to commit this change to the playground, so you should revert the changes to this file.

**Review by maribethb — CHANGES_REQUESTED** [review]
Hi, @Taru-Shukla, thank you for the PR! The overall change looks good, but there's a few steps before we can merge this:

- Address the comment left in the PR review below
- Run `npm run format` and commit the changes to automatically fix formatting issues
- Follow the google-cla bot's instructions to sign the licensing agreement

After that, we'll be able to merge this PR. Thanks!

File: tests/playgrounds/advanced_playground.html (hunk 1 above)
Comment (Taru-Shukla) [reply tests/playgrounds/advanced_playground.html]: sure, I have made those changes

**Comment by BeksOmega** [comment]
Heya @Taru-Shukla Sorry for not getting back to you sooner :/ Are you still interested in working on this? We actually recently merged the OSD branch, so this will need to be rebased against `master` to go in.

**Comment by BeksOmega** [comment]
@Taru-Shukla I'm going to go ahead and close this as stale! But if you're still interested in working on it just let me know =)
